pkg/cluster/deploybaseresources.go

```diff
@@ -444,6 +454,60 @@ func (m *manager) setMasterSubnetPolicies(ctx context.Context) error {
 	return err
 }
 
+func (m *manager) federateIdentityCredentials(ctx context.Context) error {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil
+	}
+
+	if m.doc.OpenShiftCluster.Properties.ClusterProfile.OIDCIssuer == nil {
+		return fmt.Errorf("OIDCIssuer is nil for the cluster with ID: %s", m.doc.OpenShiftCluster.ID)
+	}
+
+	issuer := to.StringPtr((string)(*m.doc.OpenShiftCluster.Properties.ClusterProfile.OIDCIssuer))
+
+	platformWIRolesByRoleName := m.platformWorkloadIdentityRolesByVersion.GetPlatformWorkloadIdentityRolesByRoleName()
+	platformWorkloadIdentities := m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
+
+	for _, identity := range platformWorkloadIdentities {
+		identityResourceId, err := azure.ParseResourceID(identity.ResourceID)
+		if err != nil {
+			return err
+		}
+
+		platformWIRole, exists := platformWIRolesByRoleName[identity.OperatorName]
+		if !exists {
+			continue
+		}
+
+		federatedIdentityCredentialResourceName, err := m.getPlatformWorkloadIdentityFederatedCredName(identity)
+		if err != nil {
+			return err
+		}
+
+		for _, sa := range platformWIRole.ServiceAccounts {
+			_, err := m.clusterMsiFederatedIdentityCredentials.CreateOrUpdate(
+				ctx,
+				identityResourceId.ResourceGroup,
+				identityResourceId.ResourceName,
+				federatedIdentityCredentialResourceName,
```

Comment:
This looks to me like it's going to overwrite the same fed cred repeatedly rather than creating multiple fed creds when a platform MI has more than one SA. Maybe the SA name should be incorporated into the fed cred name?

Reply:
I think it shouldn't overwrite, [federate-multiple-kubernetes-service-accounts-with-a-single-identity](https://azure.github.io/azure-workload-identity/docs/faq.html#how-to-federate-multiple-kubernetes-service-accounts-with-a-single-identity)

Reply:
You can definitely create multiple fed creds using the same MI, but you're going to need to give each fed cred resource a unique name.

Reply:
Given that we are limited to 120 characters, I will let @cadenmarchese @rajdeepc2792 and @kimorris27 to suggest the naming convention to be maintained :) before I make any changes. My suggestion would look something like below
```
sa := strings.Split(ServiceAccount, ":")
saSecondPart := sa[1] //second part of service account

return fmt.Sprintf("%s-%s", m.doc.ID, concat(saSecondPart, identityResourceId.ResourceName)[:83])
// 36 chars from doc.ID 
// +1 from "-"
// Remaining 83 comes from first 83 chars of concat result of saSecondPart & ResourceName
```

Reply:
What's the rationale behind using the doc ID instead of the cluster name / resource name?

Reply:
Is `ServiceAccount` a k8s SA? e.g. `system:serviceaccount:namespace:name`?

Reply:
@stevekuznetsov yes, but in this case it's in namespace:name format only: https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/RP-Config?path=/deploy/prod-config.yaml&version=GBmaster&line=2964&lineEnd=2965&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents 

So an example would be `openshift-cloud-controller-manager:cloud-controller-manager`

Reply:
OK, please consider that the name of the SA is allowed to be 246 characters long - account for this in your algorithm. Consider base32(sha224(inputs)) as a simple way to derive a short name deterministically without having cascades of length adjustment code.

Reply:
@stevekuznetsov - In the case where we aren't able to clean up the federated Identity during deletion process as [mentioned here](https://github.com/Azure/ARO-RP/pull/3847#discussion_r1779115235), how will the customer be able to find the Hashed name [base32(sha224(inputs))] to clean up identities on their own? Should the customer list all the federated identities under a given User Assigned Identity and delete them?
cc - @cadenmarchese @kimorris27 @rajdeepc2792

Reply:
updated